controller/list_controller.php

```diff
@@ -29,85 +29,111 @@ public function ideas_list($sort)
 			throw new http_exception(404, 'IDEAS_NOT_AVAILABLE');
 		}
 
-		// Build the breadcrumb off the $sort parameter
-		$breadcrumb = (in_array($sort, array(ideas::SORT_NEW, ideas::SORT_TOP, ideas::SORT_IMPLEMENTED)) ? $sort : array());
+		// Overwrite the $sort parameter if the url contains a sort query.
+		// This is needed with the sort by options form at the footer of the list.
+		$sort = ($this->request->is_set('sort')) ? $this->request->variable('sort', ideas::SORT_NEW) : $sort;
 
+		// Get additional query values the url may contain
+		$sort_direction = $this->request->variable('sd', 'd');
+		$status = $this->request->variable('status', 0);
+		$start = $this->request->variable('start', 0);
+
+		// Store original query params for use in breadcrumbs & pagination
+		$u_sort = $sort;
+		$u_status = $status;
+		$u_sort_direction = $sort_direction;
+
+		// convert the sort direction to ASC or DESC
+		$sort_direction = ($sort_direction === 'd') ? 'DESC' : 'ASC';
+
+		// If sort by "new" we really use date
 		if ($sort === ideas::SORT_NEW)
 		{
 			$sort = ideas::SORT_DATE;
 		}
 
-		$sort_direction = ($this->request->variable('sd', 'd')) === 'd' ? 'DESC' : 'ASC';
-		$status = $this->request->variable('status', 0);
-		$author = $this->request->variable(ideas::SORT_AUTHOR, 0);
-		$start = $this->request->variable('start', 0);
-
+		// if sort by "implemented", sort ideas with status implemented by date
 		if ($sort === ideas::SORT_IMPLEMENTED)
 		{
 			$status = ideas::STATUS_IMPLEMENTED;
 			$sort = ideas::SORT_DATE;
 		}
 
-		$where = ($author) ? 'idea_author = ' . (int) $author : '';
+		// Set the status name for displaying in the template
+		$status_name = (!$status && $sort == ideas::SORT_TOP) ? $this->language->lang('TOP_IDEAS') : $this->ideas->get_status_from_id($status);
 
-		if ($sort == ideas::SORT_TOP)
-		{
-			$status_name = $this->language->lang('TOP_IDEAS');
-		}
-		else
+		// For special case where we want to request ALL ideas,
+		// including the statuses normally hidden from lists.
+		if ($status === -1)
 		{
-			$status_name = $this->ideas->get_status_from_id($status);
+			$status = array(
+				ideas::STATUS_NEW,
+				ideas::STATUS_PROGRESS,
+				ideas::STATUS_IMPLEMENTED,
+				ideas::STATUS_INVALID,
+				ideas::STATUS_DUPLICATE,
+			);
+
+			$status_name = $this->language->lang('ALL_IDEAS');
 		}
 
 		// Generate ideas
-		$ideas = $this->ideas->get_ideas($this->config['posts_per_page'], $sort, $sort_direction, $status, $where, $start);
+		$ideas = $this->ideas->get_ideas($this->config['posts_per_page'], $sort, $sort_direction, $status, $start);
```

Comment:
At this point all the variables got from the request are unsanitized, are they checked later?